geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java

```diff
@@ -165,8 +165,8 @@ protected void waitForPrimary(PartitionedRegion.RetryTimeKeeper retryTimer) {
                     new RegionEventImpl(localPrimaryBucketRegion, Operation.REGION_CLEAR, null,
                         false, partitionedRegion.getMyId(), regionEvent.getEventId());
                 localPrimaryBucketRegion.cmnClearRegion(bucketRegionEvent, false, true);
+                clearedBuckets.add(localPrimaryBucketRegion.getId());
```

Comment:
How is clearedBuckets used? You are changing the code to only add a bucket to it if the size of the bucket is > 0. I can see how clearedBuckets is used down at line 184 to decide if we will increment stats. This does not seem right. If we find a bunch of empty bucket then we still did a clear and took some time so it seems wrong to not even change the stats.
Also we return clearedBuckets from this method and I can't tell what the callers do with it.
Also this check that the primary bucket is not empty does not seem safe to me. Currently if evict destroy is configured on a PR then the primary and secondary can be out of sync. So the primary could be empty but the secondary could still have entries.

Reply:
We definitely should remove the "if (localPrimaryBucketRegion.size() > 0) {"